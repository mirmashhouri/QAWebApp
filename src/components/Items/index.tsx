import { memo } from "react";
import Item from "../Item";
import Tooltip from "../Tooltip";
import { Question } from "../../services/questionService";
import styles from "./styles.module.scss";

interface Props {
  items: Question[];
  onDelete: (data: Question) => void;
  onEdit: (data: Question) => void;
}

const Items = ({ items, onDelete, onEdit }: Props) => (
  <div className={styles.items}>
    <Tooltip content="Here are all of questions">
      <h4>Question List</h4>
    </Tooltip>
    <ul>
      {items.length === 0 ? (
        <div>there is no question</div>
      ) : (
        items.map((item, index) => {
          const id = `q${index}`;
          return (
            <Item key={id} item={item} onDelete={onDelete} onEdit={onEdit} />
          );
        })
      )}
    </ul>
  </div>
);

export default memo(Items);
