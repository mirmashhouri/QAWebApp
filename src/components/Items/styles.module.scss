@import "../../theme/variables";

.items {
  border-bottom: 1px solid $color-white;
  color: $color-white;
  margin: 10px 0;
  padding: 5px;
  a {
    color: $color-white;
  }
}
.items ul {
  list-style-type: none;
  padding-left: 0;
}
