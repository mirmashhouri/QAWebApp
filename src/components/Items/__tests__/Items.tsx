import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Items from "..";

describe("<Items />", () => {
  const onDelete = () => () => undefined;
  const onEdit = () => () => undefined;
  it("renders", () => {
    const tree = render(
      <MemoryRouter>
        <Items
          items={[
            {
              question: "What is your name?",
              answer: "My name is Abbas",
            },
          ]}
          onDelete={onDelete()}
          onEdit={onEdit()}
        />
      </MemoryRouter>
    ).container.firstChild;

    expect(tree).toMatchSnapshot();
  });
});
