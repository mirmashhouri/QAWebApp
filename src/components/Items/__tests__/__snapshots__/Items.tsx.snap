// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> renders 1`] = `
<div
  class="items"
>
  <div
    class="tooltip-wrapper"
  >
    <h4>
      Question List
    </h4>
  </div>
  <ul>
    <div
      class="item"
    >
      <ul>
        <li>
          <div
            class="row"
          >
            <div
              class="col-7 col-md-9"
            >
              <button
                class="w-100 text-left"
                type="button"
              >
                What is your name?
              </button>
              <div
                class="answer "
              >
                My name is Abbas
              </div>
            </div>
            <div
              class="col-5 col-md-3 action"
            >
              <button
                type="button"
              >
                Edit
              </button>
              <span>
                 | 
              </span>
              <button
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </ul>
</div>
`;
